src/js/program/media-controller.js

```diff
@@ -25,6 +33,7 @@ export default class MediaController {
         mediaModelState.duration = duration;
         model.setProvider(provider);
         model.setMediaModel(mediaModel);
+        model.mediaController = this;
```

Comment:
Is this a good idea considering `model.mediaController` is used in other places and expects `mediaController` not to change? I think we should try to only set static properties on `model`. Anything else should be a model attribute.